rust/hyperlane-base/src/server/eigen_node.rs

```diff
@@ -0,0 +1,303 @@
+    }
+
+    /// Method to return the NodeInfo data
+    pub async fn node_info_handler() -> impl IntoResponse {
+        let node_info = NodeInfo {
+            node_name: "Hyperlane Validator".to_string(),
+            spec_version: "0.1.0".to_string(),
+            node_version: "0.1.0".to_string(),
+        };
+        Json(node_info)
+    }
+
+    /// Method to return the NodeInfo data
+    /// if signed_checkpoint - observed_checkpoint <= 1 return 200 - healthy
+    /// else if observed_checkpoint - signed_checkpoint <= 10 return 203 - partially healthy
+    /// else return 503 - unhealthy
+    pub async fn node_health_handler(core_metrics: Arc<CoreMetrics>) -> impl IntoResponse {
+        let observed_checkpoint = core_metrics
+            .latest_checkpoint()
+            .with_label_values(&["validator_observed", "ethereum"])
+            .get();
+        println!("latest_checkpoint: {}", observed_checkpoint);
+        let signed_checkpoint = core_metrics
+            .latest_checkpoint()
+            .with_label_values(&["validator_processed", "ethereum"])
+            .get();
+
+        // logic to check if the node is healthy
+        if observed_checkpoint - signed_checkpoint <= 1 {
+            // 200 - healthy
+            StatusCode::OK
+        } else if observed_checkpoint - signed_checkpoint <= 10 {
+            // 206 - partially healthy
+            StatusCode::PARTIAL_CONTENT
+        } else {
+            // 503 - unhealthy
+            StatusCode::SERVICE_UNAVAILABLE
+        }
+    }
+
+    /// Method to return a list of services
+    /// NB: hardcoded for now
+    pub async fn node_services_handler() -> impl IntoResponse {
+        let services = vec![
+            Service {
+                id: "hyperlane-validator-indexer".to_string(),
+                name: "indexer".to_string(),
+                description: "indexes the messages from the origin chain mailbox".to_string(),
+                status: ServiceStatus::Up,
+            },
+            Service {
+                id: "hyperlane-validator-submitter".to_string(),
+                name: "submitter".to_string(),
+                description: "signs messages indexed from the indexer".to_string(),
+                status: ServiceStatus::Down,
+            },
+        ];
+        Json(services)
+    }
+
+    /// Method to return the health of a service
+    pub async fn service_health_handler(service_id: String) -> impl IntoResponse {
+        // TODO: implement logic to check if the service is healthy
+        // now just return 200
+        format!("Health check for service: {}", service_id);
+        StatusCode::OK
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use axum::http::StatusCode;
+    use prometheus::Registry;
+    use serde_json::Value;
+
+    #[tokio::test]
+    async fn test_eigen_node_api() {
+        let core_metrics =
+            Arc::new(CoreMetrics::new("dummy_relayer", 37582, Registry::new()).unwrap());
```

Comment:
nit: just to avoid confusion as this is testing validator-specific behavior, change to dummy_validator here and throughout

Reply:
fixed